butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/conditions/java/JavaMatch.java

```diff
@@ -19,7 +19,9 @@
  * if they all are true. If the specified Java class file contains
  * more than one type, only the outer one will be considered
  * during evaluation. If it has none, the evaluation will result
- * in false and a warning be returned.
+ * in false and a warning be returned. If the Java class file is
+ * package-info file, it returns a warning with a default result as false.
```

Comment:
Given the points you made in the issue, which are very valid, let's please rephrase this to:

`If the Java class file is named package-info.java, the file content will not be evaluated and false and a warning will be returned. The boolean result for package-info.java file is configurable though.`